src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,47 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a person's contact information in the address book.
+ * 
+ */
+abstract class Contact {
+	private final String VALIDATION_REGEX;
+	private final String MESSAGE_CONSTRAINTS; 
+	
+	public final String value;
+	private boolean isPrivate;
+	
+	Contact(String validation_regex, String message_constraints, String val, boolean isPriv) throws IllegalValueException{
+		VALIDATION_REGEX = validation_regex; 
+		MESSAGE_CONSTRAINTS = message_constraints;
+		isPrivate = isPriv;
+		if (!isValid(val)) {
```

Comment:
Please fix the indentation and change the tab to spaces in Eclipse, see [here](http://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs)